### ISSUE TITLE
PDF reader: Underlining

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#2460]
https://forums.zotero.org/discussion/95399/feature-request-underline-option-rather-than-highlight-for-pdfs

Would this be hard to support? Is this considered a different annotation type, or is just a variation on a highlight?

People do seem to use it.

**Comment by mrtcode** [comment]
It is a different annotation type. If this is only about rendering a line instead of a rect, we could maybe do this in a bit hacky way by introducing a flag in annotation.position. But the cleanest way, of course, would be an individual annotation.

**Comment by dstillman** [comment]
Yeah, it doesn't really matter at all to us how we represent it internally, but for, say, the API, it's a little hacky to not call it an `underline` annotation type…

**Comment by mrtcode** [comment]
Underline annotations currently are only imported as "external" annotations, but when trying to transfer them with "Import Annotations…" they remain locked and stay in the PDF file.

https://github.com/zotero/pdf-worker/blob/ecef376e92257be777cfb72e3ea28904ed9d2b98/src/annotations/common.js#L65

**Comment by mrtcode** [comment]
I don't remember if we discussed that, but why not just add a pref that would make highlights to look like underlines? Wouldn't that mostly satisfy the people who want this? The difference between highlight and underline is just psychological…

**Comment by dstillman** [comment]
I'm sure there are people who use both for different things.

**Comment by dstillman** [comment]
E.g.:

> I am in favor of an underline tool in addition to a florescent highlight color. Oftentimes, I would highlight a keyword, and underline the description of it with the same color. It helps distinguish the keywords and key points from the sentences that further support them. 

https://forums.zotero.org/discussion/comment/402384/#Comment_402384

**Comment by ryanwwest** [comment]
> I'm sure there are people who use both for different things.

One example - on black-and-white ereader screens, it's hard to distinguish different colors. You might be limited to just black and 1-2 shades of grey tops. But with underlining, you can easily distinguish them since the shape distinguishes them. That's also why other types of underlining (e.g. squiggly, dashed), may be helpful for a wider variety  of devices.

**Comment by rafetkavak** [comment]
Adding strikethrough as well might be a good option.

**Comment by dstillman** [comment]
Added in Zotero 7

**Comment by ryanwwest** [comment]
Thank you!
